Source/CarthageKit/Xcode.swift

```diff
@@ -790,10 +804,36 @@ public func buildScheme(scheme: String, withConfiguration configuration: String,
 		var argsForBuilding = argsForLoading
 		argsForBuilding.onlyActiveArchitecture = .No
 
-		var buildScheme = xcodebuildTask("build", argsForBuilding)
-		buildScheme.workingDirectoryPath = workingDirectoryURL.path!
+		func fetchDestination() -> SignalProducer<String?, ReactiveTaskError> {
+			if sdk == .iPhoneSimulator {
+				let destinationLookup = TaskDescription(launchPath: "/usr/bin/xcrun", arguments: [ "simctl", "list", "devices" ])
+				return launchTask(destinationLookup)
+					|> ignoreTaskData
+					|> map { data in
+						let string = NSString(data: data, encoding: NSStringEncoding(NSUTF8StringEncoding))!
+						let regex = NSRegularExpression(pattern: "-- iOS [0-9.]+ --\\n.*?\\(([0-9A-Z]{8}-([0-9A-Z]{4}-){3}[0-9A-Z]{12})\\)", options: nil, error: nil)!
+						let lastDeviceResult = regex.matchesInString(string as String, options: nil, range: NSRange(location: 0, length: (string as NSString).length)).last as? NSTextCheckingResult
```

Comment:
Because `string` is already `NSString`, the `(string as NSString)` cast is unnecessary.